src/middlewared/middlewared/plugins/snapshot.py

```diff
@@ -95,11 +95,13 @@ async def do_create(self, data):
         """
         Create a Periodic Snapshot Task
 
-        Create a Periodic Snapshot Task that will take snapshots of specified `dataset` at specified `schedule`.
+        Create a Periodic Snapshot Task that will take snapshots of specified `dataset` or `zvol` at specified `schedule`.
```

Comment:
This is a parameter name, should remain unchanged.

For ZFS perspective everything is a dataset, what we call dataset is a dataset of FILESYSTEM type, zvol is a dataset of VOLUME type.

Reply:
The target audience isn't people who think technically about ZFS and its developer terminology. It's people who use the web UI, and in the web UI we universally do distinguish datasets and volumes that way. So stating only datasets here strongly implies "not volumes" to an end user.

Reply:
This is the API, text not available to end users. There is no `zvol` attribute in the API.

Reply:
Damn. I wish I understood the codebase better :) Id like to offer direct PRs not just ask others to do the work. would that be okay?

Reply:
Any contribution in actual code will always be very welcome ;)
As long as you're comfortable receiving harsh feedback of people with OCD :D

Reply:
And as long as you're okay reviewing rubbish PRs from people who don't really have a clue! :)

Reply:
I can approve as soon as you revert that line ; )

Reply:
Reverted. Don't know github well enough to see it, but it' not letting me re-revert it so hopefully it agrees its done. If not please do fix.